### ISSUE TITLE
[Feature Request] Native MAX Serving Support for MiniCPMForCausalLM Architecture

### ISSUE THREAD

**Issue by PMukund219** [modular/mojo#6733]
### What is your request?

I would like to add native support for the `MiniCPMForCausalLM` architecture
(model type `minicpm`). I have an implementation ready and would like to open
a PR for it, filing this issue first per maintainer guidance.
After support is added, users should be able to serve the model with:
```
max serve --model-path Goekdeniz-Guelmez/MiniCPM-2B-dpo-bf16-safetensors
```

### What is your motivation for this change?

MiniCPM-2B is a small (2.4B param), Apache 2.0-licensed dense text model
from OpenBMB with standard multi-head attention (36 heads). It is a
useful addition to MAX's supported model coverage for lightweight instruct-tuned
deployment.

### Any other details?

This is for inference support.

The new architecture follows this layout:
```
minicpm/
├── layers/
│   ├── __init__.py
│   ├── attention.py
│   ├── rope.py
│   └── transformer_block.py
├── __init__.py
├── arch.py
├── BUILD.bazel
├── minicpm.py
├── model_config.py
├── model.py
└── weight_adapters.py
```

Key config values (from HuggingFace `config.json`):
- `hidden_size`: 2304
- `num_hidden_layers`: 40
- `intermediate_size`: 5760
- `vocab_size`: 122753
- `max_position_embeddings`: 4096
- `rms_norm_eps`: 1e-5
- `hidden_act`: silu
- `rope_scaling`: null
- `scale_emb`: 12 
- `scale_depth`: 1.4 
- `dim_model_base`: 256 

Notable implementation details:
- **Depth scaling**: both attention and MLP residual branches are scaled by `scale_depth / sqrt(num_hidden_layers)` before being added back to the residual stream.
- **Embedding scaling**: embedding outputs are multiplied by `scale_emb`.
- **Logits scaling**: logits are divided by `hidden_size / dim_model_base` (equivalent to scaling the LM head input, since the head has no bias).
- **Checkpoint format**: the official `openbmb/` checkpoints ship weights exclusively as `pytorch_model.bin`. Since MAX only supports `safetensors` and `gguf`, we are targeting the community safetensors-converted checkpoint [`Goekdeniz-Guelmez/MiniCPM-2B-dpo-bf16-safetensors`](https://huggingface.co/Goekdeniz-Guelmez/MiniCPM-2B-dpo-bf16-safetensors), which is weight-identical to the official DPO checkpoint.

Useful links:
- HuggingFace model (official): https://huggingface.co/openbmb/MiniCPM-2B-dpo-bf16
- Safetensors checkpoint used for testing: https://huggingface.co/Goekdeniz-Guelmez/MiniCPM-2B-dpo-bf16-safetensors
